cogs/tags.py

```diff
@@ -107,115 +170,338 @@ async def all(self, ctx: commands.Context):
+            return self.bot.dispatch(
+                "tag_update_response",
+                message,
+                approved,
+                user=event.member,
+            )
+
+    @commands.Cog.listener()
+    async def on_tag_create_response(self, message: discord.Message, approved, user):
+        text = message.embeds[0].description
+        name = message.embeds[0].fields[0].value
+        creator_id = int(message.embeds[0].fields[1].value)
+
+        if approved:
+            tag = Tag(
+                bot=self.bot,
+                guild_id=message.guild.id,
+                creator_id=creator_id,
+                name=name,
+                text=text,
+            )
+
+            await tag.post()
+
+            author = await self.bot.resolve_user(creator_id)
+
+            return await message.edit(
+                embed=discord.Embed(
+                    title="Tag Created",
+                    description=text,
+                    timestamp=datetime.utcnow(),
+                    color=discord.Color.green(),
+                )
+                .add_field(name="Name", value=name)
+                .set_author(name=author, icon_url=author.avatar_url)
+                .set_footer(text=f"Approved By: {user}", icon_url=user.avatar_url)
+            )
+
+        await message.edit(
+            embed=discord.Embed(
+                title="Tag Creation Denied",
+                description=text,
+                timestamp=datetime.utcnow(),
+                color=discord.Color.red(),
+            )
+            .add_field(name="Name", value=name)
+            .set_author(name=user, icon_url=user.avatar_url)
+            .set_footer(text=f"Denied By: {user}", icon_url=user.avatar_url)
+        )
+
+    @commands.Cog.listener()
+    async def on_tag_update_response(self, message: discord.Message, approved, user):
+        text = message.embeds[0].description[11:]
+        name = message.embeds[0].fields[1].value
+        creator_id = int(message.embeds[0].fields[2].value)
+        before = message.embeds[0].fields[0].value
+
+        if approved:
+            tag = await self.bot.db.get_tag(guild_id=message.guild.id, name=name)
+            author = await self.bot.resolve_user(creator_id)
+
+            if tag is None:
+                return await message.edit(
+                    embed=discord.Embed(
+                        title="Tag Update Failed",
+                        description="Tag has been deleted",
+                        timestamp=datetime.utcnow(),
+                        color=discord.Color.red(),
+                    )
+                    .add_field(name="Name", value=name)
+                    .set_author(name=author, icon_url=author.avatar_url)
+                )
+
+            await tag.update(text=text)
+            return await message.edit(
+                embed=discord.Embed(
+                    title="Tag Updated",
+                    description=f"**Before**: {before}\n\n**After**: {text}",
```

Comment:
How long can the embed description be? Because both the before and after can be 2048